platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/Collectors.java

```diff
@@ -183,9 +182,15 @@ private static <T> void visitValue(Action<? super ValueSupplier.ExecutionTimeVal
 
     public static class ElementsFromCollection<T> implements Collector<T> {
         private final Iterable<? extends T> value;
+        private final SideEffect<? super Iterable<? extends T>> sideEffect;
```

Comment:
:thinking: I wonder if a decorating `Collector` that holds the side effect would be a cleaner approach.

Reply:
Good idea, I went in that direction and introduced `FixedValueCollector`. That way I can also avoid weird equals/hashcode.